test/integration/imagechange_buildtrigger_test.go

```diff
@@ -528,5 +544,8 @@ func TestMultipleImageChangeBuildTriggers(t *testing.T) {
 		if updatedConfig.Spec.Triggers[tc.triggerIndex].ImageChange.LastTriggeredImageID != "registry:5000/openshift/"+tc.name+":"+tc.tag {
 			t.Fatalf("Expected imageID equal to pull spec, got %#v", updatedConfig.Spec.Triggers[0].ImageChange)
 		}
+
+		ignoreBuilds[newBuild.Name] = ""
```

Comment:
Why use a map of strings?

Empty structs is more idiomatic, since we're only interested in the map as a poor-man's set:

```go
ignoreBuilds := make(map[string]struct{})
// ...
ignoreBuilds[newBuild.Name] = struct{}{}
```

And I'd also suggest putting this assignment in the same place where we check if the build name is in the map, to keep related code together:

```go
			build = event.Object.(*buildapi.Build)
			if _, seen := buildsSeen[build.Name]; !seen {
				buildsSeen[build.Name] = struct{}{}    // not seen before, but now it has been seen
				break
			}
```